### PR TITLE
[#31] [iOS] [UI] As a user, I can see a list of surveys on the home screen

### LINKED ISSUE #31: [iOS] [UI] As a user, I can see a list of surveys on the home screen
## Why

Once logged in, users can navigate through a list of surveys that they can take. The mobile app will show the basic info of the survey as they scroll through each survey.

## Acceptance Criteria

- Implement a horizontal scroll view with a snapping behavior
- Display the background image view
- Display the Page indicators
- Display the Title label
    - The maximum number of lines is 4
- Display the Description label
    - The maximum number of lines is 2
- Display the Survey Disclosure button

## Design

Image

[Home](https://www.figma.com/embed?embed_host=notion&url=https%3A%2F%2Fwww.figma.com%2Fproto%2FGjRPOjDyZ6f4EDL3wKarRK%2FChallenge-Mobile-App%3Fpage-id%3D0%253A1%26node-id%3D30%253A46%26scaling%3Dmin-zoom)

### PR DESCRIPTION
**Pull request by suho**
- Resolves #31

## What happened 👀

Implemented the home screen with a survey lists view.

## Insight 📝

From the beginning, I tried with the `TabView` component instead, but after checking the Figma, I decided to go with adding left right gesture, then update the content based on user actions.

## Proof Of Work 📹

https://user-images.githubusercontent.com/19943832/202677751-4617cb81-b12d-4a8f-8ccf-426b44fd132e.mp4

### CODE REVIEW

**Review by nkhanh44 — COMMENTED** [review]
So good implementation of detecting Drag gesture for the horizontal scroll view with a snapping behavior 👍 . LGTM.

Just a question. Do you think we should check if user logged in, so the next time we could navigate directly to the home screen instead of the login screen like current implementation? If I miss something, please feel free to correct me. Thank you ❤️ .

File: ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import SwiftUI
+
+extension DragGesture.Value {
+
+    enum Direction {
+
+        case left
+        case right
+        case up
+        case down
```
Comment (nkhanh44) [inline ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift:18]: This is optional. As we don't have any story that will need `up` and `down` directions. I think we could safely remove these 😄 .

File: ios/Surveys/Sources/Presentation/Views/PageControl/PageControl.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    func updateUIView(_ uiView: UIPageControl, context: Context) {
+        uiView.currentPage = currentPage
+    }
+}
+
+extension PageControl {
+
+    class Coordinator: NSObject {
+        var control: PageControl
```
[the remark this answers is not in the document]
Comment (suho) [reply ios/Surveys/Sources/Presentation/Views/PageControl/PageControl.swift]: Fixed in https://github.com/suho/kmm-ic/pull/104/commits/9f37c221277be5382fcf08b6b0a11ae8d74bcaf2

File: ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+}
+
+extension DragGesture.Value {
+
+    func detectDirection(_ tolerance: Double = 24) -> Direction? {
```
[the remark this answers is not in the document]
Comment (suho) [reply ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift]: Fixed in https://github.com/suho/kmm-ic/pull/104/commits/9f37c221277be5382fcf08b6b0a11ae8d74bcaf2

File: ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift (hunk 1 above)
Comment (suho) [reply ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift:18]: Actually, we will have the pull to refresh feature on the home screen, so having the down direction would help later. As a result, I prefer to keep all directions :D

**Comment by suho** [comment]
@nkhanh44 Currently, we don't have the sign-out feature, so I prefer to keep the current approach. I will improve later when having the sign-out feature

File: ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        gesture(
+            DragGesture()
+                .onEnded { value in
+                    guard let direction = value.detectDirection() else { return }
+                    if direction == .left || direction == .right {
+                        action(direction)
+                    }
+                }
+        )
+    }
```
Comment (vnntsu) [inline ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift:45]: what do you think about?
```suggestion
    func swipe(_ directions: Direction... = [.left, .right, .up, .down], @escaping (DragGesture.Value.Direction) -> Void) -> some View {
        gesture(
            DragGesture()
                .onEnded { value in
                    guard let direction = value.detectDirection() else { return }
                    if directions.contains(direction) {
                        action(direction)
                    }
                }
         )
    }
```

**Review by vnntsu — APPROVED** [review]
The rest LGTM! 🎉

File: ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift (hunk 4 above)
Comment (suho) [reply ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift:45]: Nice one! I will improve this :+1:

Comment (suho) [reply ios/Surveys/Sources/Supports/Extensions/SwiftUI/DragGesture+Direction.swift]: Fixed in https://github.com/suho/kmm-ic/pull/104/commits/bf55879f2088a84ab0b4de1ef3d6ce39fb2dabfe

**Pull request merged**